### PR TITLE
Benchmark tweaks

### PR DESCRIPTION
**Pull request by martinthomson**
1. Use NSS Release
2. Add workflow_dispatch
3. Move some env values up in the file

### CODE REVIEW

**Review by larseggert — APPROVED** [review]

**Comment by larseggert** [comment]
YAML error: https://github.com/mozilla/neqo/actions/runs/7925911670

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -28,7 +33,6 @@ jobs:
 
       - name: Configure Rust
         run: |
-          echo "RUSTFLAGS=-C link-arg=-fuse-ld=lld -C link-arg=-Wl,--no-rosegment" >> "$GITHUB_ENV"
           cargo install flamegraph
```
Comment (larseggert) [inline .github/workflows/bench.yml:36]:
```suggestion
        run: cargo install flamegraph
```

File: .github/workflows/bench.yml [hunk 2]
```diff
@@ -60,7 +63,6 @@ jobs:
       - name: Prepare machine
         run: |
           sudo /root/bin/prep.sh
```
Comment (larseggert) [inline .github/workflows/bench.yml:65]:
```suggestion
        run: sudo /root/bin/prep.sh
```

**Review by larseggert — COMMENTED** [review]
Minor tweaks.

File: .github/workflows/bench.yml [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
   workflow_call:
+  workflow_dispatch:
 env:
   CARGO_PROFILE_BENCH_BUILD_OVERRIDE_DEBUG: true
   CARGO_PROFILE_RELEASE_DEBUG: true
   CARGO_TERM_COLOR: always
   RUST_BACKTRACE: 1
   TOOLCHAIN: nightly
+  RUSTFLAGS:-C link-arg=-fuse-ld=lld -C link-arg=-Wl,--no-rosegment
+  PERF_CMD=record -o perf.data -F997 --call-graph lbr -g
```
Comment (larseggert) [inline .github/workflows/bench.yml:12]:
```suggestion
  PERF_CMD: record -o perf.data -F997 --call-graph lbr -g
```

File: .github/workflows/bench.yml [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 on:
   workflow_call:
+  workflow_dispatch:
 env:
   CARGO_PROFILE_BENCH_BUILD_OVERRIDE_DEBUG: true
   CARGO_PROFILE_RELEASE_DEBUG: true
   CARGO_TERM_COLOR: always
   RUST_BACKTRACE: 1
   TOOLCHAIN: nightly
+  RUSTFLAGS:-C link-arg=-fuse-ld=lld -C link-arg=-Wl,--no-rosegment
```
Comment (larseggert) [inline .github/workflows/bench.yml:11]:
```suggestion
  RUSTFLAGS: -C link-arg=-fuse-ld=lld -C link-arg=-Wl,--no-rosegment
```

**Review by larseggert — CHANGES_REQUESTED** [review]
I think you need to change at least `=` -> `:` in the `env:` clause to unbreak this.

File: .github/workflows/bench.yml [hunk 5]
```diff
@@ -28,7 +33,6 @@ jobs:
 
       - name: Configure Rust
         run: |
```
Comment (martinthomson) [inline .github/workflows/bench.yml:35]:
```suggestion
        run: cargo install flamegraph
```

File: .github/workflows/bench.yml (hunk 1 above)
Comment (martinthomson) [inline .github/workflows/bench.yml:36]:
```suggestion
```

File: .github/workflows/bench.yml [hunk 6]
```diff
@@ -58,9 +61,7 @@ jobs:
 
       # Disable turboboost, hyperthreading and use performance governor.
       - name: Prepare machine
-        run: |
-          sudo /root/bin/prep.sh
-          echo "PERF_CMD=record -o perf.data -F997 --call-graph lbr -g" >> "$GITHUB_ENV"
+        run: sudo /root/bin/prep.sh
 
       # Pin the benchmark run to core 0 and run all benchmarks at elevated priority.
       - name: Run cargo bench
```
Comment (martinthomson) [inline .github/workflows/bench.yml:67]: https://github.com/mozilla/neqo/actions/runs/7959004190/job/21725490360?pr=1663#step:10:24 shows that nice isn't doing anything.

Comment (larseggert) [reply .github/workflows/bench.yml:67]: Yes, it's on my list to check. I did give the bench user rights in limits.conf.

**Review by larseggert — APPROVED** [review]

**Comment by martinthomson** [comment]
@larseggert Is there some way to set a looser constraint on coverage regressions?  A failure due to a 0.01% decrease is pretty silly.

**Pull request merged**